cli/parameter.go

```diff
@@ -26,23 +27,39 @@ type workspaceParameterFlags struct {
 }
 
 func (wpf *workspaceParameterFlags) allOptions() []serpent.Option {
-	options := append(wpf.cliBuildOptions(), wpf.cliParameters()...)
+	options := append(wpf.cliEphemeralParameters(), wpf.cliParameters()...)
 	options = append(options, wpf.cliParameterDefaults()...)
 	return append(options, wpf.alwaysPrompt())
 }
 
-func (wpf *workspaceParameterFlags) cliBuildOptions() []serpent.Option {
+func (wpf *workspaceParameterFlags) cliEphemeralParameters() []serpent.Option {
 	return serpent.OptionSet{
+		// Deprecated - replaced with ephemeral-parameter
 		{
 			Flag:        "build-option",
 			Env:         "CODER_BUILD_OPTION",
 			Description: `Build option value in the format "name=value".`,
-			Value:       serpent.StringArrayOf(&wpf.buildOptions),
+			UseInstead:  []serpent.Option{{Flag: "ephemeral-parameter"}},
+			Value:       serpent.StringArrayOf(&wpf.ephemeralParameters),
 		},
+		// Deprecated - replaced with prompt-ephemeral-parameters
 		{
 			Flag:        "build-options",
 			Description: "Prompt for one-time build options defined with ephemeral parameters.",
-			Value:       serpent.BoolOf(&wpf.promptBuildOptions),
+			UseInstead:  []serpent.Option{{Flag: "prompt-ephemeral-parameters"}},
+			Value:       serpent.BoolOf(&wpf.promptEphemeralParameters),
+		},
+		{
+			Flag:        "ephemeral-parameter",
+			Env:         "CODER_EPHEMERAL_PARAMETER",
+			Description: `Define ephemeral parameters in the format "name=value".`,
```

Comment:
```suggestion
			Description: `Set the value of ephemeral parameters defined in the template. The format is "name=value".`,
```
I felt this was a bit vague, as if new parameters unrelated to the template could be defined. This is my take but can likely be improved further.